modules/motions/BottleTwoCatchAction.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * @file   BottleTwoCatchAction.cpp
+ * @brief  ボトル2つ目のキャッチ動作クラス
+ * @author nishijima515
+ */
+
+#include "BottleTwoCatchAction.h"
+
+// コンストラクタ
+BottleTwoCatchAction::BottleTwoCatchAction(
+    Robot& _robot, double _forwardDistance, double _idsSpeed, double _ultrasonicDistance,
+    double _udclSpeed, double _angle, double _rotatePower, const PidGain& _pidGain,
+    const CameraServer::BoundingBoxDetectorRequest& _detectionRequest)
+  : CompositeMotion(_robot),
+    forwardDistance(_forwardDistance),
+    idsSpeed(_idsSpeed),
+    ultrasonicDistance(_ultrasonicDistance),
+    udclSpeed(_udclSpeed),
+    angle(_angle),
+    rotatePower(_rotatePower),
+    pidGain(_pidGain),
+    detectionRequest(_detectionRequest)
+{
+}
+
+void BottleTwoCatchAction::run()
+{
+  // ボトル探索開始時点の走行距離を取得する
+  double initialRightMotorCount = robot.getMotorControllerInstance().getRightMotorCount();
+  double initialLeftMotorCount = robot.getMotorControllerInstance().getLeftMotorCount();
+  double initialDistance = Mileage::calculateMileage(initialRightMotorCount, initialLeftMotorCount);
+
+  double initialAngle = robot.getIMUControllerInstance().getAngle();  // 動作開始時の角度
+
+  SocketClient& client = robot.getSocketClient();
+  CameraServer::BoundingBoxDetectorResponse response;
+  bool success = client.executeLineDetection(detectionRequest, response);
+
+  // PCIDSを動かす。ボトル探索。
+  PictureColorDistanceStraight pcids(robot, forwardDistance, idsSpeed, PidGain(0.08, 0.02, 0.05),
+                                     detectionRequest);
+  pcids.run();
+
+  UltrasonicDistanceCameraLineTrace udcl(robot, ultrasonicDistance, forwardDistance, udclSpeed, 400,
+                                         pidGain, detectionRequest);
+
+  udcl.run();
+  // stop.run();
+
+  // 動作終了時点の走行距離を取得する
+  double currentRightMotorCount = robot.getMotorControllerInstance().getRightMotorCount();
+  double currentLeftMotorCount = robot.getMotorControllerInstance().getLeftMotorCount();
+  double currentDistance = Mileage::calculateMileage(currentRightMotorCount, currentLeftMotorCount);
+
+  double totalAngleToTurn = robot.getIMUControllerInstance().getAngle();
```

Comment:
ここで、initialAngleを使う予定だった？
```suggestion
  double totalAngleToTurn = robot.getIMUControllerInstance().getAngle() - initialAngle;
```